script/model/type.py

```diff
@@ -54,6 +39,51 @@ class OpUnit(enum.IntEnum):
     UPDATE = 24,
     DELETE = 25,
     CREATE_INDEX = 26,
+    CREATE_INDEX_MAIN = 27,
+    PARALLEL_MERGE_HASHJOIN = 28,
+    PARALLEL_MERGE_AGGBUILD = 29,
+    PARALLEL_SORT_STEP = 30,
+    PARALLEL_SORT_MERGE_STEP = 31
+
+
+class ExecutionFeature(enum.IntEnum):
+    # Debugging information
+    QUERY_ID = 0,
+    PIPELINE_ID = 1,
+
+    # # features
+    NUM_FEATURES = 2,
+    FEATURES = 3,
+
+    # input features
+    EXEC_MODE = 4,
+    EST_OUTPUT_ROWS = 5,
+    KEY_SIZES = 6,
+    NUM_KEYS = 7,
+    EST_CARDINALITIES = 8,
+    MEM_FACTOR = 9,
+    NUM_LOOPS = 10,
+    NUM_CONCURRENT = 11,
+
+    # interval input features
+    TXNS_DEALLOCATED = 12,
+    TXNS_UNLINKED = 13,
+    BUFFER_UNLINKED = 14,
+    READONLY_UNLINKED = 15,
+    INTERVAL = 16,
+
+    # outputs
+    START_TIME = 17,
+    CPU_ID = 18,
+    CPU_CYCLES = 19,
+    INSTRUCTIONS = 20,
+    CACHE_REF = 21,
+    CACHE_MISS = 22,
+    REF_CPU_CYCLES = 23,
+    BLOCK_READ = 24,
+    BLOCK_WRITE = 25,
+    MEMORY_B = 26,
+    ELAPSED_US = 27
```

Comment:
If it's not too hard to refactor, can we still have another `Target` enum class for all of these? They should not really be called features (*labels*, rather) and they're not specific to the execution engine either. I think it's more accurate if we can separate them.

Reply:
Separated!